### PR TITLE
feat: load employee statuses on demand

### PR DESCRIPTION
**Pull request by djaiss**
The employee statuses, on the employee profile page, have some problems.

1. The query to fetch them is done all the time, even if the employee can’t change statuses.
2. They use a previous concept of "collections" that I implemented two years ago, but it’s a really bad idea as a concept, for a lot of reasons.

In #436, I’m changing how employee statuses work under the hood, and I need the improvements I've done in this PR for this.

I wanted to split the work in smaller PR so #436 is not too big.

In short, this PR
* loads employee statuses on demand on the Employee profile page
* is the first step to enable the removal of the EmployeeStatusCollection method.

Note: ideally, all the data in the header of an Employee profile page should be loaded asynchronously, but we have to do it slowly, one by one, and proceed carefully.

### CODE REVIEW

**Comment by djaiss** [comment]
I have to merge this in order to move forward with my other PR!
